components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/DefaultUserStoreAttributeMappingParser.java

```diff
@@ -0,0 +1,172 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.user.store.configuration.utils;
+
+import org.apache.axiom.om.OMAttribute;
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.impl.builder.StAXOMBuilder;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.user.store.configuration.model.UserStoreAttributeDO;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLStreamException;
+
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.ATTRIBUTE_ID;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.CLAIM_CONFIG;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.CLAIM_URI;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.DIALECT;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.DIALECTS;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.DISPLAY_NAME;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.LOCAL_DIALECT_URL;
+
+/**
+ * Parser to read default user store mappings from claim-config.xml.
+ */
+public class DefaultUserStoreAttributeMappingParser {
+
+    private static Map<String, UserStoreAttributeDO> defaultAttributes;
+    private static final Log LOG = LogFactory.getLog(DefaultUserStoreAttributeMappingParser.class);
+
+    private DefaultUserStoreAttributeMappingParser() {
+
+        init();
+    }
+
+    private static final class ParserHolder {
+
+        static final DefaultUserStoreAttributeMappingParser PARSER = new DefaultUserStoreAttributeMappingParser();
+    }
+
+    public static DefaultUserStoreAttributeMappingParser getInstance() {
+
+        return ParserHolder.PARSER;
+    }
+
+    private static void init() {
+
+        InputStream inStream = null;
+
+        File attributeMappingXml = new File(CarbonUtils.getCarbonConfigDirPath(),
+                CLAIM_CONFIG);
+        try {
+            if (attributeMappingXml.exists()) {
+
+                inStream = new FileInputStream(attributeMappingXml);
```

Comment:
You need to make sure to close the InputStream once you are done. Better to use try-with-resources here.

Reply:
fixed